### ISSUE TITLE
Extract shell and file-system functions from NextJS template

### ISSUE THREAD

**Issue by mikebarkmin** [openpatch/hyperbook#387]
If we can extract the shell (Layout, Navigation, Drawer and Toc) from the NextJS template into its own package, we could use the shell in the VS Code extension for providing a full preview of the site. We could also add a docsify-like mode (client-side-rendering), which just relies on a single HTML file, which fetches markdown files from a remote. This would mean we only need to generate a navigation.json. This should make the build time almost instant. If we want to do that, we probably need to extract file-system related function into their own package.

To summarize we need:

- A shell package
- A fs package
- A csr template and extend the hyperbook cli (e.g.: hyperbook build --csr / --single-file)
- Add a full preview to the VS Code Extension
